src/Cmfd.cpp

```diff
@@ -362,198 +338,191 @@ void Cmfd::computeXS() {
+
+/**
+ * @brief Compute the surface diffusion coefficient for a given cmfd cell,
+ *        cell surface, and group.
  * @details This method uses finite differencing to compute the surface
- *         diffusion coefficients (\f$ \hat{D} \f$) for each cell surface.
- *         Additionally, the surface diffusion coefficent correction factors
- *         (\f$ \tilde{D} \f$) are computed in order to conserve the net
- *         leakage out of each mesh cell. This serves to preserve both local
- *         and global neutron balance.
+ *         diffusion coefficient (\f$ \hat{D} \f$) or surface diffusion
+ *         coefficient correction (\f$ \tilde{D} \f$) for a given cmfd cell, cell
+ *         surface, and cmfd energy group. If the MOC iteration is zero,
+ *         (\f$ \tilde{D} \f$) is returned as zero. Since (\f$ \hat{D} \f$) and
+ *         (\f$ \tilde{D} \f$) are dependent on each other, they must be computed
+ *         together; therefore, the boolean correction is used to indicate which
+ *         value is to be returned.
+ * @param cmfd_cell A Cmfd cell
+ * @param surface A surface of the Cmfd cell
+ * @param group A Cmfd energy group
  * @param moc_iteration MOC iteration number
+ * @param correction Boolean indicating whether (\f$ \hat{D} \f$) or
+ *                   (\f$ \tilde{D} \f$) is to be returned
+ * @return The surface diffusion coefficient, (\f$ \hat{D} \f$) or
+ *         (\f$ \tilde{D} \f$)
  */
-void Cmfd::computeDs(int moc_iteration) {
-
-  log_printf(INFO, "Computing CMFD diffusion coefficients...");
+FP_PRECISION Cmfd::getSurfaceDiffusionCoefficient(int cmfd_cell, int surface,
+                                                  int group, int moc_iteration,
+                                                  bool correction) {
 
-  FP_PRECISION d, d_next, d_hat, d_tilde;
-  FP_PRECISION flux, flux_next, f, f_next;
+  FP_PRECISION dif_surf, dif_surf_corr;
   FP_PRECISION current, current_out, current_in;
-  FP_PRECISION length, length_perpen, next_length_perpen;
-  FP_PRECISION sense;
-  int next_surface;
-  int cell_id, cell_next_id;
-
-  /* Loop over mesh cells in y direction */
-  #pragma omp parallel for private(d, d_next, d_hat, d_tilde, current, flux, \
-    flux_next, f, f_next, length, length_perpen, next_length_perpen, \
-    sense, next_surface, cell_id, cell_next_id, current_in, current_out)
-  for (int y = 0; y < _num_y; y++) {
-
-    /* Loop over Mesh cells in x direction */
-    for (int x = 0; x < _num_x; x++) {
+  FP_PRECISION flux_next;
+
+  /* Get diffusivity and flux for Mesh cell */
+  FP_PRECISION dif_coef = getDiffusionCoefficient(cmfd_cell, group);
+  FP_PRECISION flux = _old_flux->getValue(cmfd_cell, group);
+  int cmfd_cell_next = getCellNext(cmfd_cell, surface);
+  FP_PRECISION delta_interface = getSurfaceWidth(surface);
+  FP_PRECISION delta = getPerpendicularSurfaceWidth(surface);
+  int sense = getSense(surface);
+
+  /* Correct the diffusion coefficient with Larsen's effective diffusion
```

Comment:
Is there a good reference for Larsen's effective diff coeff? If so, you might include it here or in the doxygen comment.